documentation/modules/exploit/multi/http/sugarcrm_webshell_cve_2023_22952.md

```diff
@@ -0,0 +1,193 @@
+## Vulnerable Application
+
+This module exploits CVE-2023-22952, a Remote Code Execution (RCE) vulnerability in SugarCRM 11.0 Enterprise,
+Professional, Sell, Serve, and Ultimate versions prior to `11.0.5` and SugarCRM 12.0 Enterprise, Sell, and
+Serve versions prior to `12.0.2`.
+
+The vulnerability occurs due to a lack of appropriate validation when uploading a malicious `PNG` file with
+embedded PHP code to the `/cache/images/` directory on the web server using the vulnerable endpoint
+`/index.php?module=EmailTemplates&action=AttachFiles`. Once uploaded to the server, depending on server configuration,
+the attacker can access the malicious PNG file via HTTP or HTTPS, thereby executing the malicious PHP code and
+gaining access to the system.
+
+This vulnerability does not require authentication because there is a missing authentication check in the
+`loadUser()` method in `include/MVC/SugarApplication.php`. After a failed login, the session does not get
+destroyed and hence the attacker can continue to send valid requests to the application. See this
+[AttackerKB Article](https://attackerkb.com/topics/E486ui94II/cve-2023-22952) for more details.
+
+Because of this, any remote attacker, regardless of authentication, can exploit this vulnerability to gain
+access to the underlying operating system as the user that the web services are running as (typically `www-data`).
+
+Installing a vulnerable test bed requires a Linux machine with the vulnerable SugarCRM software loaded.
```

Comment:
We should probably expand these install instructions. At the very least a download link would be good to provide.

Reply:
see https://github.com/rapid7/metasploit-framework/pull/17507/commits/290618fa668b5d924031fd6ba0a96bb998cef6c3

Reply:
I can't make heads or tails of where I'm meant to download from these instructions. There is mention of a download manager here but creating an account and trying to go to https://auth.sugarcrm.com/dashboard shows no products, and using the download manager also returns errors as there are no products assigned to our account.

Reply:
This is the error I get for reference:

```
HTTP request returned status code 404: {"error":"not_found","error_message":"Could not find record or user does not have access to: downloads in module: Accoun (truncated...)
```

Reply:
I can't make heads or tails of where I'm meant to download from these instructions. There is mention of a download manager here but creating an account and trying to go to https://auth.sugarcrm.com/dashboard shows no products, and using the download manager also returns errors as there are no products assigned to our account.

This is the error I get for reference:

```
HTTP request returned status code 404: {"error":"not_found","error_message":"Could not find record or user does not have access to: downloads in module: Accoun (truncated...)
```